WordPress/gutenberg
# Add content schema to pattern editing view

## What?
Closes #59815

Adds a new 'content' panel when editing patterns. When a user names a block to declare it as overridable, that block will now display in the panel.

Questions:
- Is the 'Content' title for the panel right? I felt it's probably the way to go as it helps create a connection with the 'content' panel that shows for pattern block instances, but I'm open to other opinions.
- Is it worth adding some 'help' text. E.g. 'Blocks that display here can be overridden when this pattern is inserted'?
- Should the panel be above the others (Revisions, Categories)?
- Should there be an empty state? At the moment the panel doesn't render when there are no named blocks, but it could do, and it might be a nice call to action to name blocks. It may be good to consider this in conjunction with #59813.

## How?
- Adds a new private `ContentPanel` component in the `patterns` package, with the various UI pieces
- Adds a new private `PatternContentPanel` component in the `editor` package, which determines when the panel should be shown (whether a `wp_block` is being edited)
- Render those components in the post and site editors
- Moves the `isOverridableBlock` function to the patterns package, which acts as the source of truth for whether a block has overrides (which is important as this will likely be changing with [Use metadata.name only as the hint for pattern overrides](https://github.com/WordPress/gutenberg/pull/59956#top))

## Testing Instructions
1. Create a new pattern and edit it or edit an existing one (this should work in the post or site editors)
2. Add paragraph, heading, image or button blocks to the pattern and name them
3. Go to the 'Pattern' tab in the inspector and observe the new 'Content' panel
4. Click on one of the blocks in the 'Content' panel to edit it

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/677833/16a7e8a8-8d19-409e-ab9d-01f94ffbe00d
